File: Java/lab17/src/main/java/org/example/Mediator.java
package org.example;

import java.util.ArrayList;
import java.util.List;

public class Mediator implements IMediator{
    private final List<ConsultantAbstract> consultantList=new ArrayList<>();
    @Override
    public void addConsultant(ConsultantAbstract consultant){
        consultantList.add(consultant);
    }
    @Override
    public void sendOrder(Order order,Consultant consultant) {
        for(ConsultantAbstract c:consultantList){
            if(c!=consultant){
                c.getOrder(order);
            }
        }
    }
}

File: Python/lab17/Mediator.py
from IMediator import IMediator


class Mediator(IMediator):
    def __init__(self):
        self.consultant_list = []

    def add_consultant(self, consultant):
        self.consultant_list.append(consultant)

    def send_order(self, order, consultant):
        for c in self.consultant_list:
            if c != consultant:
                c.get_order(order)
